port/stack_trace.cc

```diff
@@ -135,6 +137,68 @@ void PrintStack(void* frames[], int num_frames) {
 }
 
 void PrintStack(int first_frames_to_skip) {
+#ifdef ROCKSDB_DLL
+  // LIB_MODE=shared build produces modiocre information from the above
+  // backtrace+addr2line stack trace method. Try to use GDB in that case.
+  bool dll = true;
+#else
+  bool dll = false;
+#endif
+  // Also support invoking interactive debugger on stack trace, with this
+  // envvar set to non-empty
+  char* debug_env = getenv("ROCKSDB_DEBUG");
+  bool debug = debug_env != nullptr && strlen(debug_env) > 0;
+
+  if (dll || debug) {
+    // Allow ouside debugger to attach, even with Yama security restrictions
+#ifdef PR_SET_PTRACER_ANY
+    (void)prctl(PR_SET_PTRACER, PR_SET_PTRACER_ANY, 0, 0, 0);
+#endif
+    // Try to invoke GDB, either for stack trace or debugging
+    char pid_str[20];
+    snprintf(pid_str, sizeof(pid_str), "%lld", (long long)getpid());
```

Comment:
Btw, I don't think heap allocation is supported in signal handlers, which is why we use stack-allocated buffers and mostly plain C.